### ISSUE TITLE
Approved Merge Requests (EE-only) is not persisted

### ISSUE THREAD

**Issue by DieBauer** [jenkinsci/gitlab-plugin#711]
### Context
- **Gitlab plugin version**: 1.5.3
- **Gitlab version**: 10.3.5
- **Jenkins version**: 2.89.3
- **Job type**: Freestyle

### Problem description

The plugin doesn't respect a flag for 
<f:entry title="Approved Merge Requests (EE-only)" field="triggerOnApprovedMergeRequest">

When toggling this setting and applying the change on refresh of the job, the toggle is activated again. 

It should persist the toggle. 

There are no log statements in the Jenkins log.

**Comment by shark-horse** [comment]
I too cannot disable "Approved Merge Requests (EE-only)".

What does that setting do? Link to documentation saying what it does would be welcome.

I was trying to disable it because of a hunch that it means to trigger a build _upon an approval_. Basically I was trying to troubleshoot #633 .

Context
Gitlab plugin version: 1.5.3
Gitlab version: 10.4.0-ee
Jenkins version: 2.89.3
Job type: Maven

**Comment by prawen** [comment]
Hello,

I'm facing the same issue. Will there be any fix for this?

**Comment by eddie3716** [comment]
I too am experiencing the same problem.  I found the discussion https://github.com/jenkinsci/gitlab-plugin/issues/633 and followed it until I found this new open issue.

Gitlab plugin version: 1.5.3
Gitlab version: 10.2.2-ee
Jenkins version: 2.89.3

**Comment by eddie3716** [comment]
@jakub-bochenski @Argelbargel  the change made from #664 is the cause of this bug.

**Comment by jakub-bochenski** [comment]
Sorry, I cannot reproduce that. Here is a screenshot of me changing this value.

![screenshot_20180214_175933](https://user-images.githubusercontent.com/2794994/36217098-e2d09f88-11b0-11e8-8840-9e43fcb4ae16.png)

A bug that I can reproduce is connected to the Web form used to configure the job. 
Because the trigger object lacks a getter for the value it will always display true regardless of the actual setting.
I recommend checking the config xml to be sure.

**Comment by shark-horse** [comment]
@jakub-bochenski Are you saying you _can_ change the value somehow, but not through the Web UI? If so, how?

Are you using a plugin to keep track of project configuration history? How do you get to that view in the screenshot you shared?

I'm just editing through the UI currently:
![image](https://user-images.githubusercontent.com/30937134/36269184-1f232ed2-123e-11e8-893b-9144a25359d1.png)

We're using a plugin to backup our configuration via a git repo. I may be able to submit a change via that, but I think additional steps are required for the change to take effect. Really, I don't know how to control the configuration via direct file manipulation--IME there's more to it than meets the eye, maybe because of the aforementioned plugin.

**Comment by shark-horse** [comment]
Looking through history, I do see a change to the config.xml:

```       <triggerOnPipelineEvent>false</triggerOnPipelineEvent>
       <triggerOnAcceptedMergeRequest>false</triggerOnAcceptedMergeRequest>
       <triggerOnClosedMergeRequest>false</triggerOnClosedMergeRequest>
-      <triggerOnApprovedMergeRequest>true</triggerOnApprovedMergeRequest>
+      <triggerOnApprovedMergeRequest>false</triggerOnApprovedMergeRequest>
       <triggerOpenMergeRequestOnPush>source</triggerOpenMergeRequestOnPush>
       <triggerOnNoteRequest>true</triggerOnNoteRequest>
```

but Jenkins is still building upon Approval, which is incorrect for our workflow.

**Comment by jakub-bochenski** [comment]
> but Jenkins is still building upon Approval, which is incorrect for our workflow.

I haven't checked that, but it would be a separate issue from the incorrect display in Web UI.

We use job-dsl plugin to manage the job configuration, that's why I haven't noticed the UI issue before.

The change above was done via Web UI though, the screenshot comes from the JobConfigHistory plugin.

**Comment by williamzhao87** [comment]
I am also seeing this issue.  Has anyone found any workaround?

**Comment by jakub-bochenski** [comment]
@williamzhao87 there is a PR fixing it linked above :(

**Comment by lambourn** [comment]
Any update on this issue? I also experience this - as soon as a MR gets approved, the pipeline gets kicked of again. In our workflow we run only want to run builds when pushes are done to the MR but not on approval.

**Comment by shark-horse** [comment]
FWIW, as a workaround, I configured the Gitlab plugin (in the Jenkins job) to _not_ build on the creation of an MR, which means we have to wait until approval for a build. (We have configured Gitlab to only allow merging with at least 1 approval + a successful build, i.e. a failed or pending build will block merging.)

**Comment by dskrvk** [comment]
@shark-horse I'm curious how you configured Gitlab to block on Jenkins builds - is it through Gitlab CI? Did you also have to configure CI runners?

**Comment by shark-horse** [comment]
@dskrvk What I was talking about in my previous comment was a config change in a Jenkins job, not a change in Gitlab. Specifically, I configured the Gitlab plugin in Jenkins to _not_ build on creation of an MR:
![image](https://user-images.githubusercontent.com/30937134/37602312-13fd44ea-2b5a-11e8-9096-f911d9ce9164.png)

Notice that `Opened Merge Request Events` is unchecked (because `Approved Merge Requests (EE-only)` is always checked.)

In Gitlab, I am using a Webhook that triggers on Merge Request Events (gitlabUrl/settings/integrations). I know it is sending information to Jenkins every time the MR is _opened_ because, prior to disabling builds on opened MR's, Jenkins would build on opened MR's. Jenkins is simply ignoring that per the configuration in the Jenkins job.

FWIW:
- I opted for Webhooks instead of the Jenkins CI service because we need certain flexibility not provided by the service.
- Beyond the scope of this issue, I configured the job in Jenkins to build if someone says "Jenkins retry" in a comment, and I configured Gitlab to notify Jenkins upon a "Note event" which seems to correspond to comments in MR's.
